### ISSUE TITLE
why not using latest arcadeflex src?

### ISSUE THREAD

**Issue by georgemoralis** [philrod1/james#1]
interesting project! 
but why you don't use my latest arcadeflex src ?

https://github.com/georgemoralis/arcadeflex036

it seems you use my old google code one from arcadeflex029

**Comment by philrod1** [comment]
First of all, thank you for ArcadeFlex!

I am a writing up artificial intelligence PhD student.  My thesis work is demonstrated using the original Ms. Pac-Man code running in the emulator.  Back when I started, I went looking for a Java emulator and I found ArcadeFlex.  It had two problems, though.  It was to complex, as I only needed Ms. Pac-Man, and it wasn't Object-Oriented.  I wanted to be able to run partial emulators as a forward model for my AI, but this wasn't possible.  I decided to build an OO emulator from the ground up, using code from ArcadeFlex wherever possible, but focused specificity on Ms. Pac-Man.  So _James_ isn't a fork of ArcadeFlex---it has very different priorities.

If I ever get the time, I'd love to reintroduce support for other games.  Ideally, I would like to build a game factory class, which reads in a JSON or XML text file description of the game (driver) and returns an instance of that game machine.  Probably won't happen, though :-/

I am hoping to present my work at the CIG2016 conference.  I put the code up here---without my final AI---in case anyone else is interested in AI for Ms. Pac-Man.

**Comment by georgemoralis** [comment]
Very interesting project really like it.

I like the way you made arcadeflex more OOP , was my first try also , but due to size of the project it was abandonded early.
I have also wrote a c to java convertor to convert drivers of mame to arcadeflex compatible code it's inside arcadeflex 0.36 you can check it (a more advance is on my new repo 0.37b11 one)

If you prepare you presentation i would love to see it :)

**Comment by philrod1** [comment]
Yes, only having to deal with a single architecture, and not even properly implementing memory mapped IO, made this much easier to do.  Trying to be as complete as MAME is a mammoth task.

GitHub no longer supports private messages, but you can email me at phil@core-its.co.uk.  That's an old address, but I can send you my proper email and edit this post later.  I don't want my private email open to the public.
